theano/gpuarray/tests/test_dnn.py

```diff
@@ -1067,6 +1067,63 @@ def get_conv3d_test_cases():
     return itt
 
 
+def run_conv_small_batched_vs_multicall(inputs_shape, filters_shape, batch_sub):
+    # Function to check issue #5985 (see tests below): https://github.com/Theano/Theano/issues/5985
+
+    # Error occurs with algorithm `small` (CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM)
+    algo = 'small'
+
+    batch_size = inputs_shape[0]
+
+    utt.seed_rng()
+    inputs_val = np.random.random(inputs_shape).astype('float32')
+    filters_val = np.random.random(filters_shape).astype('float32')
+    # Scale down the input values to prevent very large absolute errors
+    # due to float rounding
+    inputs_val /= 10
+    filters_val /= 10
+    inputs = theano.shared(inputs_val)
+    filters = theano.shared(filters_val)
+
+    if len(inputs_shape) == 5:
+        dnn_func = dnn.dnn_conv3d
+    else:
+        dnn_func = dnn.dnn_conv
+    conv = dnn_func(img=inputs, kerns=filters, algo=algo)
+    # Just compute first and last outputs, to reduce execution time.
+    sub_conv_top = dnn_func(img=inputs[:batch_sub], kerns=filters, algo=algo)
+    sub_conv_bottom = dnn_func(img=inputs[(batch_size - batch_sub):], kerns=filters, algo=algo)
+    f = theano.function([], [conv, sub_conv_top, sub_conv_bottom], mode=mode_with_gpu)
+    res_all, res_batch_top, res_batch_bottom = f()
+    for i in range(batch_sub):
+        # Check first ouputs.
+        utt.assert_allclose(res_batch_top[i], res_all[i])
+        # Then check last outputs.
+        p = batch_size - batch_sub + i
+        # It seems there is a limit batch size of 65536  with algorithm `small`.
+        checked_limit = 2**16
+        if p >= checked_limit:
+            # It seems results are repeated in the entire conv.
+            # It should not happen.
+            if np.allclose(res_all[p % checked_limit], res_all[p]):
+                print('\nconv[%d] == conv[%d] == %s' % (p % checked_limit, p, res_all[p]))
+        utt.assert_allclose(res_batch_bottom[i], res_all[p])
+
+
+def test_batched_conv_small():
+    yield (run_conv_small_batched_vs_multicall, (65534, 2, 2, 2), (1, 2, 2, 2), 5)  # OK
```

Comment:
this is in each PR tests run. To help keep it fast, only test the size 65536 and 65537 in both test.